research/conditional/utils/layer_manager.py

```diff
@@ -32,16 +32,31 @@ def __init__(
 
     def _register_layers(self, model):
```

Comment:
It assumes some block naming conventions. Do we have them written down somewhere?

Reply:
i have not, i will attempt to put it down in a docstring